### ISSUE TITLE
Refactoring SPARQLStore as a subclass of SQLStore

### ISSUE THREAD

**Issue by simontaurus** [SemanticMediaWiki/SemanticMediaWiki#5748]
While `ElasticStore` subclasses `SQLStore` 
https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/c8f778a85aca38d4d78173c4014f9dc40d067473/src/Elastic/ElasticStore.php#L39

`SPARQLStore` has a private member that is a `SQLStore` 
https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/c8f778a85aca38d4d78173c4014f9dc40d067473/src/SPARQLStore/SPARQLStore.php#L49-L66

Why not making `SPARQLStore` an actual subclass of `SQLStore`? This would solve at least https://github.com/SemanticMediaWiki/SemanticResultFormats/issues/827

**Comment by JeroenDeDauw** [comment]
Sounds plausible (I would need to investigate for a more specific opinion)

Immediate thoughts: SQLStore is big and inheriting a lot of concrete code (1) is terrible software design and (2) might come with complications. Using types that are defined by small interfaces is better, though might be hard to achieve due to existing approaches in the code.

**Comment by simontaurus** [comment]
> Using types that are defined by small interfaces is better

I agree. In this case the simpliest solution is to make the `$baseStore` member public (see https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/5749) in order to access it when we need the underlying SQLStore (see https://github.com/SemanticMediaWiki/SemanticResultFormats/pull/861)
